### PR TITLE
Apply min_itemsize to index even when not appending

### LINKED ISSUE #10381: BUG: DataFrame.to_hdf doesn't pass along min_itemsize for index
Unless I'm seeing something wrong

``` python
In [21]: df = DataFrame(dict(A = 'foo', B = 'bar'),index=range(5)).set_index("A")

In [22]: df.to_hdf('store.h5', 'test', format='table', min_itemsize={'index': 10})

In [23]: store = pd.HDFStore('store.h5')

In [24]: store.get_storer('test').table
Out[24]:
/test/table (Table(5,)) ''
  description := {
  "index": StringCol(itemsize=3, shape=(), dflt=b'', pos=0),
  "values_block_0": StringCol(itemsize=3, shape=(1,), dflt=b'', pos=1)}   # <---- I think this should be 10
  byteorder := 'irrelevant'
  chunkshape := (10922,)
  autoindex := True
  colindexes := {
    "index": Index(6, medium, shuffle, zlib(1)).is_csi=False}
```

and FYI this raises (not sure if it should work)

``` python
In [25]: df.index.name = 'theindex'

In [26]: df.to_hdf('store.h5', 'test2', format='table', min_itemsize={'theindex': 10})
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
```

Just a report right now... no time.

### PR DESCRIPTION
**Pull request by toobaz**
- [x] closes #10381
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/io/tests/test_pytables.py [hunk 1]
```diff
@@ -1372,6 +1372,22 @@ def check_col(key, name, size):
                          min_itemsize={'index': 4})
             tm.assert_series_equal(store.select('ss2'), df['B'])
 
+            # min_itemsize in index of DataFrame.to_hdf() (GH 10381)
```
Comment (toobaz) [inline pandas/io/tests/test_pytables.py]: OK, replaced with ``store.*``

**Comment by jreback** [comment]
can you add a replica of @TomAugspurger example as well (you can make a new test function is fine). Just to cover the bases.

**Comment by jreback** [comment]
otherwise lgtm.

**Comment by toobaz** [comment]
Appveyor is stuck again

**Comment by jreback** [comment]
rebase again, problem with appeveyor which i just fixed

**Pull request closed without merging**